Lib/unittest/mock.py

```diff
@@ -3003,6 +3005,59 @@ def __set__(self, obj, val):
         self(val)
 
 
+class ThreadingMock(MagicMock):
+    """
+    A mock that can be used to wait until on calls happening
+    in a different thread.
+    """
+
+    def __init__(self, *args, **kwargs):
+        _safe_super(ThreadingMock, self).__init__(*args, **kwargs)
+        self.__dict__["_event"] = threading.Event()
+        self.__dict__["_expected_calls"] = []
+        self.__dict__["_events_lock"] = threading.Lock()
```

Comment:
I wonder if it might be pragmatic to prefix these with `_mock_` to further reduce the chance of conflicting with child attributes of the mocked class?